src/ol/renderer/canvas/VectorTileLayer.js

```diff
@@ -673,10 +666,12 @@ class CanvasVectorTileLayerRenderer extends CanvasTileLayerRenderer {
     const tiles = this.renderedTiles;
     const clips = [];
     const clipZs = [];
+    let ready = true;
     for (let i = tiles.length - 1; i >= 0; --i) {
       const tile = /** @type {import("../../VectorRenderTile.js").default} */ (
         tiles[i]
       );
+      ready = ready && !tile.getReplayState(layer).dirty;
```

Comment:
`ready` is true when the currently drawn tiles have no loading icons. Tile state is tracked separately.